### ISSUE TITLE
Modeling communication systems on Mars

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1412]
**Is your feature request related to a problem? Please describe.**
- Settlers can't develop settlements on Mars if they don't have good and functional communication systems in place.

**Describe the rationale**
 - A lot of good thoughts have been given to the design of a communication system and a messaging system on Mars in the past. See this [wiki](https://github.com/mars-sim/mars-sim/wiki/Communication-Channels).
 
**Describe the goal(s) you have in mind**
- Define how and when a communication channel is being used.
- Define the range, the bandwidth, the power requirement and the frequency of a communication channel.
- Define both the software and the hardware side of a communication system.
- Create a list of (both mobile and stationary) communication equipments that compose a communication system.
- Rework the Command or Communication function for a building 
- Rework what a Command and Control building would serve in a settlement

**Describe the non-goal(s) associated with the goal(s)**
- Terrestrial communication is the main focus of discussion in this thread. Once the foundational design work is done, we will consider space communication and satellite communication.

**Specifications**
- post 3.8.0
